pisa-proxy/app/config/src/config.rs

```diff
@@ -21,185 +21,208 @@ use proxy::proxy::{MySQLNode, MySQLNodes, ProxiesConfig, ProxyConfig};
+        }
     }
 
-    pub fn get_mysql_nodes(&self) -> &Vec<MySQLNode> {
-        &self.mysql_nodes
+    pub fn build_from_env(mut self) -> Self {
+        self._deployed_ns = 
+            env::var("PISA_DEPLOYED_NAMESPACE").unwrap_or(PISA_PROXY_DEFAULT_DEPLOYED_NAMESPACE.to_string());
+        self._deployed_name =
+            env::var("PISA_DEPLOYED_NAME").unwrap_or(PISA_PROXY_DEFAULT_DEPLOYED_NAME.to_string());
+        self._pisa_svc = 
+            env::var("PISA_CONTROLLER_SERVICE").unwrap_or(PISA_CONTROLLER_DEFAULT_SERVICE.to_string());
+        self._pisa_ns = 
+            env::var("PISA_CONTROLLER_NAMESPACE").unwrap_or(PISA_CONTROLLER_DEFAULT_NAMESPACE.to_string());
+        self._pisa_host  =
+            env::var("PISA_CONTROLLER_HOST").unwrap_or(format!("{}.{}:8080", self._pisa_svc, self._pisa_ns));
+        self._local =     
+            env::var(PISA_PROXY_CONFIG_ENV_LOCAL_CONFIG).unwrap_or("false".to_string());
+        self._git_tag =     
+            env::var(PISA_PROXY_VERSION_ENV_GIT_TAG).unwrap_or("".to_string());
+        self._git_commit =     
+            env::var(PISA_PROXY_VERSION_ENV_GIT_COMMIT).unwrap_or("".to_string());
+        self._git_branch =     
+            env::var(PISA_PROXY_VERSION_ENV_GIT_BRANCH).unwrap_or("".to_string());
+        self._http_path = format!(
+                "http://{}/apis/configs.database-mesh.io/v1alpha1/namespaces/{}/proxyconfigs/{}",
+                self._pisa_host, self._deployed_ns, self._deployed_name
+            );
+
+        self
     }
 
-    pub fn get_admin(&self) -> &Admin {
-        &self.admin
+    pub fn build_from_file(self, path: String) -> PisaProxyConfig {
+        let mut config = PisaProxyConfig::new();
+        let mut file = match File::open(path) {
+            Err(e) => {
+                eprintln!("{:?}", e);
+                std::process::exit(-1);
+            }
+            Ok(file) => file,
+        };
+
+        let mut config_str = String::new();
+        file.read_to_string(&mut config_str).unwrap();
+        config = toml::from_str(&config_str).unwrap();
+        config
     }
 
-    pub fn load_http() -> Result<Config, Box<dyn std::error::Error>> {
-        let deployed_ns = env::var("PISA_DEPLOYED_NAMESPACE")
-            .unwrap_or(PISA_PROXY_DEFAULT_DEPLOYED_NAMESPACE.to_string());
-        let deployed_name =
-            env::var("PISA_DEPLOYED_NAME").unwrap_or(PISA_PROXY_DEFAULT_DEPLOYED_NAME.to_string());
-        let pisa_svc = env::var("PISA_CONTROLLER_SERVICE")
-            .unwrap_or(PISA_CONTROLLER_DEFAULT_SERVICE.to_string());
-        let pisa_ns = env::var("PISA_CONTROLLER_NAMESPACE")
-            .unwrap_or(PISA_CONTROLLER_DEFAULT_NAMESPACE.to_string());
-        let pisa_host =
-            env::var("PISA_CONTROLLER_HOST").unwrap_or(format!("{}.{}:8080", pisa_svc, pisa_ns));
-
-        info!(
+    pub fn build_from_http(self, path: String) -> Result<PisaProxyConfig, Box<dyn std::error::Error>> {
+        print!(
```

Comment:
`print!` don't wrap,  `println!` should be used here.

Reply:
> `print!` don't wrap,  `println!` should be used here.

I should remove this print

Reply:
fixed